### DISCUSSION TITLE
OON File plan_id

### DISCUSSION THREAD

**Discussion by jamessiefke** [CMSgov/price-transparency-guide#119] [Q&A] [answered]
Can the plan_id field contain a string of HIOS values for all plans within a single state?  If so, are the values just comma separated?  

We identified that we would be providing this data by Carrier.

**Comment by ghost** [comment]
@jamessiefke Please see this thread as it would represent what you are asking for at the root schema level and could be used for OON File
Reformat Root Schema To Support Multiple Plan/HIOS/EIN and Less Individual Files #93

**Comment by jonberkery-work** [comment]
@jamessiefke,  In my opinion that would be helpful if plan_id could be redefined as an array of Strings, on both the In-Network-rates, and allowed-amount files.  Today the data model is defined as a single string field, and therefore as you probably recognize, requires you to now recreate the same file X times over, the only difference between the files being the different "header information" in most cases (plan_name, plan_id_type, plan_id, plan_market_type, etc....)

I would be in favor of this recommendation.

I do understand, that if this field was adjusted to be an array, then it would also provide the ability to put multiple EINs within that field also, therefore if Company A and Company B went to go get "their" file then they would see each other's EINs in a file they might have expected to be "their" own.

I wonder if a more suitable recommendation would be to both - allow for the dynamic generation of the file "header" data mentioned above, and allow the plan_id field to be an Array of Strings.  This would cut down on the redundancy of these large data files.

As background - I am not a decision maker on this schema, I work for a Health Insurer who is in your same boat, trying to build this file as efficiently as possible.

I am open to other suggestions

@tdfow 
@BobSyracuse
@shaselton 
@shaselton-usds 
@ryan-thomas-peterson

**Reply by ghost** [reply]
Our driver table for the Plans and associated HIOS/EINs has a indicator that calls out that the Plan is expecting their own "individual" file as in the case of a Self Funded group. This indicator would be used in a plan_id array scenario to create a single element array. This table is also used when we have to "fan out" the duplicate negotiated rate data that is found between plans.

**Comment by shaselton-usds** [comment, marked as answer]
Wrapping up old threads -- multiple plans can be represented in a single file now by leveraging the [table of contents file](https://github.com/CMSgov/price-transparency-guide/tree/master/schemas/table-of-contents).
